### PR TITLE
Fix Category Widget search on android

### PR DESCRIPTION
**Pull request by rjimenezda**
Fix support ticket https://github.com/CartoDB/support/issues/1074

There's no point in re-rendering the search title component when the data changes. The only data used from this model is the column, so we should re-render on such

Also I've added specs + comment so this does not happen again

### CODE REVIEW

File: lib/assets/javascripts/deep-insights/widgets/category/title/search-title-view.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ module.exports = CoreView.extend({
     this.model.lockedCategories.bind('change add remove', this.render, this);
     this.add_related_model(this.model.lockedCategories);
 
-    this.dataviewModel.bind('change:data', this.render, this);
```
Comment (rubenmoya) [inline lib/assets/javascripts/deep-insights/widgets/category/title/search-title-view.js]: I'd try to take advantage that we're modifying this method and change the `.bind` + `ad_related_model` for `.listenTo`

Comment (rjimenezda) [reply lib/assets/javascripts/deep-insights/widgets/category/title/search-title-view.js]: Yep

File: lib/assets/javascripts/deep-insights/widgets/category/title/search-title-view.js [hunk 2]
```diff
@@ -72,20 +72,12 @@ module.exports = CoreView.extend({
   },
 
   _initBinds: function () {
-    this.model.bind('change:search', this._onSearchToggled, this);
```
Comment (rjimenezda) [inline lib/assets/javascripts/deep-insights/widgets/category/title/search-title-view.js]: @rubenmoya

**Comment by rjimenezda** [comment]
retest this please

**Pull request merged**